libzecale/circuits/fields/fp12_2over3over2_gadgets.tcc

```diff
@@ -361,6 +412,294 @@ void Fp12_2over3over2_mul_by_024_gadget<Fp12T>::generate_r1cs_witness()
     _z1z3z5_times_x0x2x4.generate_r1cs_witness();
 }
 
+// Fp12_2over3over2_mul_gadget methods
+
+template<typename Fp12T>
+Fp12_2over3over2_mul_gadget<Fp12T>::Fp12_2over3over2_mul_gadget(
+    libsnark::protoboard<FieldT> &pb,
+    const Fp12_2over3over2_variable<Fp12T> &A,
+    const Fp12_2over3over2_variable<Fp12T> &B,
+    const Fp12_2over3over2_variable<Fp12T> &result,
+    const std::string &annotation_prefix)
+    : libsnark::gadget<FieldT>(pb, annotation_prefix)
+    , _A(A)
+    , _B(B)
+    , _result(result)
+    , _a0_times_b0(
+          pb,
+          A._c0,
+          B._c0,
+          Fp6_3over2_variable<Fp6T>(pb, FMT(annotation_prefix, " a0*b0")),
+          " _a0_times_b0")
+    // result0 = a0*b0 + non_residue*a1*b1
+    //   <=> a1*b1 = (result0 - a0*b0) * non_residue.inverse
+    , _a1_times_b1(
+          pb,
+          A._c1,
+          B._c1,
+          fp6_mul_by_non_residue_inverse<Fp12T>(
+              pb,
+              _result._c0 - _a0_times_b0._result,
+              FMT(annotation_prefix, " a1*b1")),
+          FMT(annotation_prefix, " _a1_times_b1"))
```

Comment:
Since you're using Karatsuba here, let's switch the annotations to `v0` (resp. `v1`) for `a0*b0` (resp. `a1*b1`). See: https://github.com/clearmatics/zecale/pull/17#discussion_r450927132

Reply:
switched to v0, v1

Reply:
So you renamed the gadget and kept the wire name, right? I'd lean toward naming the output wire of the circuit `v1`.
We have already talked about the need to iterate/polish the annotations as part of a PR once all these a merged. Raising this here not to forget, but feel free to defer this to a later PR

Reply:
Opened a ticket for that here: https://github.com/clearmatics/zecale/issues/20

Reply:
Yes, that makes sense for the future.  My thinking is that the gadgets will eventually be renamed to `compute_<name>` in a later pass, at which point it will make sense to change the annotations to match, and then use names like `v0` etc for the transient variables that hold those values.  For now I thought it best to consistently make the annotations match the member names.